src/components/autocomplete.js

```diff
@@ -26,18 +26,18 @@ const Autocomplete = () => {
   const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(0);
   const [getCatBreeds, { loading, data }] = useLazyQuery(GET_BREED);
 
-  function changeHandler(query) {
-    if (!query) return setSuggestions([]);
-    setInput(query);
-
-    const debouncedChangeFilter = debounce(() => {
+  const debouncedApiCall = useCallback(
+    debounce((text) => {
       getCatBreeds({
-        variables: { breedName: query }
+        variables: { breedName: text }
       });
-    }, 500);
+    }, 500),
+    []
+  );
 
-    debouncedChangeFilter();
-  }
+  useEffect(() => {
+    debouncedApiCall(input);
+  }, [input, debouncedApiCall]);
 
   useEffect(() => {
     if (data && !loading) {
```

Comment:
I think this may be true if the current query returns an error, then `data` would hold the previous result. Not sure if this is what we want.